projects/mc_interface/inc/wavesculptor.h

```diff
@@ -38,29 +39,266 @@ typedef struct WaveSculptorResetCmd {
+  WAVESCULPTOR_MEASUREMENT_ID_CONTOLLER_RAIL = 9,
+  WAVESCULPTOR_MEASUREMENT_ID_FAN_SPEED = 10,
+  WAVESCULPTOR_MEASUREMENT_ID_SINK_MOTOR_TEMPERATURE = 11,
+  WAVESCULPTOR_MEASUREMENT_ID_AIR_IN_CPU_TEMPERATURE = 12,
+  WAVESCULPTOR_MEASUREMENT_ID_AIR_OUT_CAP_TEMPERATURE = 13,
+  WAVESCULPTOR_MEASUREMENT_ID_ODOMETER_BUS_AMPHOURS = 14
 } WaveSculptorMeasurementId;
 
-// Driver Controls Base Addr + 2
+// Motor Controller Base Addr + 0
+typedef struct WaveSculptorIdInfo {
+  // "TRIa" stored as a string. msg[0] = 'T', msg[1] = 'R'..
+  char tritium_id[4];
+
+  // Device serial number, allocated at manufacture.
+  uint32_t serial_number;
+} WaveSculptorIdInfo;
+static_assert(sizeof(WaveSculptorIdInfo) == 8, "WaveSculptorIdInfo is not 8 bytes");
+
+// Motor Controller Base Addr + 1
+typedef struct WaveSculptorStatusInfo {
+  // Limit flags indicate which control loop is limiting the output current of
+  // the motor controller
+  union {
+    struct {
+      uint16_t bridge_pwm : 1;
+      uint16_t motor_current : 1;
+      uint16_t velocity : 1;
+      uint16_t bus_current : 1;
+      uint16_t bus_voltage_upper : 1;
+      uint16_t bus_voltage_lower : 1;
+      uint16_t heatsink_temp : 1;
+      uint16_t reserved : 9;
+    } _PACKED;
+    uint16_t raw;
+  } limit_flags;
+
+  // Error flags indicate errors
+  union {
+    struct {
+      uint16_t hw_overcurrent : 1;
+      uint16_t sw_overcurrent : 1;
+      uint16_t dc_bus_overvoltage : 1;
+      uint16_t motor_position : 1;
+      uint16_t watchdog_reset : 1;
+      uint16_t config_read_error : 1;
+      uint16_t undervoltage_15v : 1;
+      uint16_t reserved : 9;
+    } _PACKED;
+    uint16_t raw;
+  } error_flags;
+
+  // The index of the motor currently being used
+  uint16_t active_motor;
+  uint16_t reserved;
+} WaveSculptorStatusInfo;
+static_assert(sizeof(WaveSculptorStatusInfo) == 8, "WaveSculptorStatusInfo is not 8 bytes");
+
+// Motor Controller Base Addr + 2
 typedef struct WaveSculptorBusMeasurement {
-  float bus_voltage;  // in V
-  float bus_current;  // in A
+  // Units: (V)olts
+  // DC bus voltage at the controller.
+  float bus_voltage;
+
+  // Units: (A)mps
+  // Current drawn from the DC bus by the controller.
+  float bus_current;
 } WaveSculptorBusMeasurement;
+static_assert(sizeof(WaveSculptorBusMeasurement) == 8, "WaveSculptorBusMeasurement is not 8 bytes");
 
-// Driver Controls Base Addr + 3
+// Motor Controller Base Addr + 3
 typedef struct WaveSculptorVelocityMeasurement {
+  // Units: revolutions per minute
+  // Motor angular frequency.
   float motor_velocity_rpm;
+
+  // Units: m/s
+  // Vehicle velocity.
   float vehicle_velocity_ms;
 } WaveSculptorVelocityMeasurement;
+static_assert(sizeof(WaveSculptorVelocityMeasurement) == 8,
+              "WaveSculptorVelocityMeasurement is not 8 bytes");
+
+// Motor Controller Base Addr + 4
+typedef struct WaveSculptorPhaseCurrentMeasurement {
+  // Units: (A)mps
+  // RMS current in motor Phase B.
```

Comment:
Nit: Define `RMS` Root mean square is I think what you mean, but not everyone is exposed to that.